src/Avalonia.Vulkan/VulkanDevice.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Silk.NET.Vulkan;
+using Silk.NET.Vulkan.Extensions.KHR;
+
+namespace Avalonia.Vulkan
+{
+    public class VulkanDevice : IDisposable
+    {
+        private object _lock = new object();
+
+        private VulkanDevice(Device apiHandle, VulkanPhysicalDevice physicalDevice, Vk api)
+        {
+            ApiHandle = apiHandle;
+            Api = api;
+
+            api.GetDeviceQueue(apiHandle, physicalDevice.QueueFamilyIndex, 0, out var queue);
+
+            var vulkanQueue = new VulkanQueue(this, queue);
+            Queues.Add(vulkanQueue);
+
+            PresentQueue = vulkanQueue;
+
+            CommandBufferPool = new VulkanCommandBufferPool(this, physicalDevice);
+        }
+
+        public Device ApiHandle { get; }
+        public Vk Api { get; }
+
+        public List<VulkanQueue> Queues { get; } = new();
+        public VulkanQueue PresentQueue { get; }
+        public VulkanCommandBufferPool CommandBufferPool { get; }
+
+        internal static List<string> RequiredDeviceExtensions { get; } = new() { KhrSwapchain.ExtensionName };
+
+        public void Dispose()
+        {
+            WaitIdle();
+            CommandBufferPool?.Dispose();
+            Queues.Clear();
+        }
+
+        public static unsafe VulkanDevice Create(VulkanInstance instance, VulkanPhysicalDevice physicalDevice,
+            VulkanOptions options)
+        {
+            if(options.VulkanDeviceInitialization != null)
+            {
+                return new VulkanDevice(options.VulkanDeviceInitialization.CreateDevice(instance.Api, instance.ApiHandle, physicalDevice, options), physicalDevice, instance.Api);
+            }
+            else
+                throw new Exception("VulkanDeviceInitialization is not found. Device can't be created.");
+        }
+
+        internal void Submit(CommandBuffer commandBuffer, SubmitInfo submitInfo, Fence fence = new())
```

Comment:
You are not using given `commandBuffer` (which makes sense since `SubmitInfo` should have it already).